flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -216,9 +220,10 @@
         // Lock is only taken, because #checkAvailability asserts it. It's a small penalty for
         // thread safety.
         synchronized (this.availableMemorySegments) {
-            if (checkAvailability()) {
-                availabilityHelper.resetAvailable();
-            }
+            // guarantee that we have one buffer on initialization
```

Comment:
1. Can you expand the java doc why is it important? Also maybe mention the difference between input and output?
2. Can you respect here `numberOfRequiredMemorySegments`? If this value is zero, we should probably skip requesting.
3. Can you also keep the best effort code `checkAvailability()` that was requesting all buffers at once if they were available?

Reply:
1. Added more comments.
2. The argument check already guarantees that the numberOfRequiredMemorySegments must be greater than 0.
3. From, it seem that checkAvailability only request one buffer, not requesting all buffers at once (correct me if I am wrong). Do you mean we enhance this logic and request all buffers at once?

Reply:
1. thanks
2. and 3. Yes, you are right. Sorry for causing confusion.